server/channels/api4/saml.go

```diff
@@ -135,7 +150,13 @@ func addSamlIdpCertificate(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	auditRec := c.MakeAuditRecord("addSamlIdpCertificate", audit.Fail)
-	defer c.LogAuditRec(auditRec)
+	defer func() {
+		if c.Err == nil {
+			auditRec.Success()
```

Comment:
The function calls `audit.Success` before returning. Why does it need to be called here as well?

Reply:
I guess the defer function I added is not even needed since `c.LogAuditRec(auditRec)` is expected to be called when the function exits (regardless of success or failure). It would be better it stayed at the previous state it was.  Am I correct?

Reply:
I have updated the PR based on your review.